### PR TITLE
Store gen_kw parameters using numpy files instead of block files

### LINKED ISSUE #4324: Remove enkf_node implementation of GEN_KW
All interaction should now be in python, so we can change the underlying data storage and fetching to be python only.

### PR DESCRIPTION
**Pull request by ManInFez**
**Issue**
Resolves #4324 

**Approach**
_Short description of the approach_

## Pre review checklist

- [x] Added appropriate release note label
- [x] PR title captures the intent of the changes, and is fitting for release notes.
- [x] Commit history is consistent and clean, in line with the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md).

Adding labels helps the maintainers when writing release notes. This is the [list of release note labels](https://github.com/equinor/ert/labels?q=release-notes).

### CODE REVIEW

File: src/ert/_c_wrappers/enkf/enkf_main.py [hunk 1]
```diff
@@ -416,55 +454,47 @@ def sample_prior(
...
+                            realization=realization_nr,
+                            data=vals,
+                        )
+                else:
+                    enkf_node = EnkfNode(config_node)
+                    if (
+                        not enkf_node.has_data(storage, NodeId(0, realization_nr))
+                        or current_status == STATE_LOAD_FAILURE
+                    ):
+                        rng = self.realizations[realization_nr]
```
Comment (oyvindeide) [inline src/ert/_c_wrappers/enkf/enkf_main.py:497]: Not used

File: src/ert/_c_wrappers/enkf/enkf_main.py [hunk 2]
```diff
@@ -116,7 +144,21 @@ def _generate_parameter_files(
     for key in ens_config.getKeylistFromVarType(
         EnkfVarType.PARAMETER + EnkfVarType.EXT_PARAMETER
     ):
+
         node = ens_config[key]
+        type_ = node.getImplementationType()
+        if type_ == ErtImplType.GEN_KW:
+            if node.getUseForwardInit():
```
Comment (oyvindeide) [inline src/ert/_c_wrappers/enkf/enkf_main.py:151]: Think we removed the option of having this for GEN_KW?

File: src/ert/_c_wrappers/enkf/enkf_fs.py [hunk 3]
```diff
@@ -171,6 +248,34 @@ def copy_from_case(
             nodes,
             active,
         )
+        self._copy_parameter_files(other, nodes, [i for i, b in enumerate(active) if b])
```
Comment (oyvindeide) [inline src/ert/_c_wrappers/enkf/enkf_fs.py:251]: Probably worth adding some tests to this

File: src/ert/_c_wrappers/enkf/enkf_fs.py [hunk 4]
```diff
@@ -135,6 +141,77 @@ def realizationList(self, state: "RealizationStateEnum") -> "IntVector":
...
+        if not input_path.exists():
+            raise KeyError(f"Unable to load GEN_KW for key: {key}")
+
+        np_data = np.load(input_path / f"{key}.npy")
+        with open(input_path / f"{key}-keys", "r") as f:
+            keys = [k.strip() for k in f.readlines()]
+
+        return np_data, keys
+
+    def load_gen_kw_as_dict(
```
Comment (oyvindeide) [inline src/ert/_c_wrappers/enkf/enkf_fs.py:185]: Do we need both this and `load_gen_kw`?

**Review by oyvindeide — COMMENTED** [review]
Looks like a good improvement :+1: Some comments

File: src/ert/_c_wrappers/enkf/enkf_fs.py (hunk 4 above)
Comment (ManInFez) [reply src/ert/_c_wrappers/enkf/enkf_fs.py:185]: load_gen_kw is used in es_update

File: src/ert/_c_wrappers/enkf/enkf_fs.py (hunk 3 above)
Comment (ManInFez) [reply src/ert/_c_wrappers/enkf/enkf_fs.py:251]: There exist a test which runs copy_case in test_case_tool.py

**Comment by ManInFez** [comment]
test this please!

File: src/ert/_c_wrappers/enkf/enkf_fs.py (hunk 4 above)
Comment (oyvindeide) [reply src/ert/_c_wrappers/enkf/enkf_fs.py:185]: Could it be refactored to use the same API, or is it very different?

File: src/ert/_c_wrappers/enkf/enkf_fs.py (hunk 3 above)
Comment (oyvindeide) [reply src/ert/_c_wrappers/enkf/enkf_fs.py:251]: Should perhaps have a test with multiple parameters? Think in general the test foundation is a bit shaky for multiple parameters/responses

**Review by frode-aarstad — APPROVED** [review]
Looks good 👍🏻

File: src/ert/_c_wrappers/enkf/enkf_fs.py (hunk 4 above)
Comment (ManInFez) [reply src/ert/_c_wrappers/enkf/enkf_fs.py:185]: it is quite different, but I think a reiteration of the API should be done when we have all the data

**Pull request merged**